docs/check_docstrings.py

```diff
@@ -0,0 +1,62 @@
+import logging
+import os
+import re
+import sys
+from glob import glob
+
+logger = logging.getLogger(__name__)
+
+
+def check_blank_line_above_param_section(file_path):
+    """
+    Check that every docstring with both a body section and a parameter
+    section separates the two sections with exactly one blank line. Log
+    errors and return count.
+
+    :param file_path: File path to check for error.
+    :return: Per file error count.
+    """
+    error_count = 0
+    with open(file_path, "r") as file:
+        content = file.read()
+
+    regex = (
+        r" {4,}\"\"\"\n(?:^[^:]+?[^\n])+(\n|\n\n\n+) {4,}(:p|:r)(?:.*\n)+? {4,}\"\"\""
+    )
+
+    bad_docstrings = re.finditer(regex, content, re.MULTILINE)
+    for docstring in bad_docstrings:
+        line_number = content.count("\n", 0, docstring.start()) + 1
+
+        # Log error message.
+        msg = "Must have exactly 1 blank line between docstring body and parameter sections."
+        logger.error(f"{file_path}: {line_number}: {msg}")
```

Comment:
Just reporting the failed file would have been sufficient, but this is pretty nice for the end user. Thanks

Reply:
😇